cmd/util/ledger/migrations/fix_broken_data_migration.go

```diff
@@ -0,0 +1,172 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/atree"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type FixSlabsWithBrokenReferencesMigration struct {
+	log           zerolog.Logger
+	rw            reporters.ReportWriter
+	accountsToFix map[common.Address]struct{}
+	nWorkers      int
+}
+
+var _ AccountBasedMigration = &FixSlabsWithBrokenReferencesMigration{}
+
+const fixSlabsWithBrokenReferencesName = "fix-slabs-with-broken-references"
+
+func NewFixBrokenReferencesInSlabsMigration(
+	rwf reporters.ReportWriterFactory,
+	accountsToFix map[common.Address]struct{},
+) *FixSlabsWithBrokenReferencesMigration {
+	return &FixSlabsWithBrokenReferencesMigration{
+		rw:            rwf.ReportWriter(fixSlabsWithBrokenReferencesName),
+		accountsToFix: accountsToFix,
+	}
+}
+
+func (m *FixSlabsWithBrokenReferencesMigration) InitMigration(
+	log zerolog.Logger,
+	_ []*ledger.Payload,
+	nWorkers int,
+) error {
+	m.log = log.
+		With().
+		Str("migration", fixSlabsWithBrokenReferencesName).
+		Logger()
+	m.nWorkers = nWorkers
+
+	return nil
+}
+
+func (m *FixSlabsWithBrokenReferencesMigration) MigrateAccount(
+	_ context.Context,
+	address common.Address,
+	oldPayloads []*ledger.Payload,
+) (
+	newPayloads []*ledger.Payload,
+	err error,
+) {
+
+	if _, exist := m.accountsToFix[address]; !exist {
+		return oldPayloads, nil
+	}
+
+	migrationRuntime, err := NewAtreeRegisterMigratorRuntime(address, oldPayloads)
```

Comment:
I recall some previous conversation where we said creating a migration runtime with all payloads can be very expensive (not sure how expensive that is though). Can the runtime/storage here be created using only the payloads belongs to the accounts in `m.accountsToFix`, as an optimization?